sentry-tower/src/http.rs

```diff
@@ -0,0 +1,146 @@
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+use http_::Request;
+use tower_layer::Layer;
+use tower_service::Service;
+
+/// Tower Layer that logs Http Request Headers.
+///
+/// The Layer can also optionally start a new performance monitoring transaction
+/// based on incoming distributed tracing headers.
+#[derive(Clone, Default)]
+pub struct SentryHttpLayer {
+    start_transaction: bool,
+}
+
+impl SentryHttpLayer {
+    /// Creates a new Layer that only logs Request Headers.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Creates a new Layer that also starts a new performance monitoring transaction.
+    pub fn with_transaction() -> Self {
+        Self {
+            start_transaction: true,
+        }
+    }
+}
+
+/// Tower Service that logs Http Request Headers.
+///
+/// The Service can also optionally start a new performance monitoring transaction
+/// based on incoming distributed tracing headers.
```

Comment:
i find this wording a bit ambiguous (same for the layer): is it a new transaction or continues the transaction set in the tracing headers?

Maybe: "The service can optionally set an active transaction to continue a transaction present in the request's distributed tracing headers"

Reply:
maybe the concepts here are a bit confusing. It is a new transaction, but it belongs to an existing trace. :-D

Reply:
the new wording is better 👍